resources/spotlight.rb

```diff
@@ -0,0 +1,43 @@
+resource_name :spotlight
+default_action :set
+
+property :volume, String, name_property: true
+property :indexed, [true, false], default: true
+property :searchable, [true, false], default: true
+
+action_class do
+  def state
+    new_resource.indexed ? 'on' : 'off'
+  end
+
+  def search
+    new_resource.searchable ? '' : '-d'
+  end
+
+  def target_volume
+    volume_path(new_resource.volume)
+  end
+
+  def volume_path(volume)
```

Comment:
We can use a ternary operator here, as well as using `::File.join` to avoid the string interpolation.

```ruby
def volume_path(volume)
  volume == '/' ? volume : ::File.join '/Volumes', volume
end
```